### DISCUSSION TITLE
Instrument Thanos with Exemplars

### DISCUSSION THREAD

**Discussion by yeya24** [thanos-io/thanos#3938] [Ideas & Features]
While I am looking at https://github.com/thanos-io/thanos/pull/3846 and the E2E tests for exemplars, I find that there are few applications available instrumented with exemplars and it is hard to add test cases for that pr.

As exemplars is available in Prometheus, I think we can start trying to instrument Thanos with exemplars.

So what metrics should we add exemplars to? Maybe we can start with HTTP metrics?

Feel free to leave any comments. Ideas are welcomed!

**Comment by metalmatze** [comment]
Yes, absolutely and I'd say let's start with the HTTP metrics for the query / query frontend. 
`http_request_duration_seconds` histogram for query and query_range endpoints should be really good entrypoints!
